src/iop/colorin.c

```diff
@@ -1764,6 +1764,89 @@ void reload_defaults(dt_iop_module_t *module)
     color_profile = DT_COLORSPACE_EMBEDDED_ICC;
   }
 
+  // We'll update the input profile hint with information on the embedded ICC profile if it exists.
+  // Since the tooltip info is now image dependent, we need to set the tooltip for each image change.
+
+  // We need gui_data to access widget in order to change tooltip.
+  dt_iop_colorin_gui_data_t *g = (dt_iop_colorin_gui_data_t *) module->gui_data;
+
+  // reload_defaults() can be called with unavailable (i.e., NULL) gui_data.
+  // In this case, we have nothing to do with tooltips.
+  if(!g) goto skip_tooltip_changing;
+
+  char datadir[PATH_MAX] = { 0 };
+  char confdir[PATH_MAX] = { 0 };
+  dt_loc_get_datadir(datadir, sizeof(datadir));
+  dt_loc_get_user_config_dir(confdir, sizeof(confdir));
+  char *system_profile_dir = g_build_filename(datadir, "color", "in", NULL);
+  char *user_profile_dir = g_build_filename(confdir, "color", "in", NULL);
+  char *tooltip_part_profile_dirs = g_strdup_printf(_("darktable loads external ICC profiles from\n%s\nand\n%s"),
+                                                    user_profile_dir, system_profile_dir);
+  g_free(system_profile_dir);
+  g_free(user_profile_dir);
+
+  // In case of embedded ICC profile we will modify tooltip with profile info,
+  // otherwise reset tooltip to generic text.
+  if((color_profile == DT_COLORSPACE_EMBEDDED_ICC) && g)
+  {
+    cmsHPROFILE cmsprofile = cmsOpenProfileFromMem(img->profile, img->profile_size);
+
+    char iccDesc[64]; iccDesc[0] = '\0';
+    cmsGetProfileInfoASCII(cmsprofile, cmsInfoDescription, "en", "US", iccDesc, 64);
+    char iccManuf[64]; iccManuf[0] = '\0';
+    cmsGetProfileInfoASCII(cmsprofile, cmsInfoManufacturer, "en", "US", iccManuf, 64);
+    char iccModel[64]; iccModel[0] = '\0';
+    cmsGetProfileInfoASCII(cmsprofile, cmsInfoModel, "en", "US", iccModel, 64);
+
+    // This is the only profile field in which, although it usually contains a short
+    // copyright text, in theory profile creators can put as long text as they want. So,
+    // we can't take a fast approach of statically allocating memory of some sufficient size
+    // and have to find out the size of the field at run-time and dynamically allocate memory.
+    char* iccCopyr;
+    guint32 bufsize = cmsGetProfileInfoASCII(cmsprofile, cmsInfoCopyright, "en", "US", NULL, 0);
+    if(bufsize)
+    {
+      iccCopyr = malloc(bufsize+1);
+      cmsGetProfileInfoASCII(cmsprofile, cmsInfoCopyright, "en", "US", iccCopyr, bufsize);
+    }
+    else
+    {
+      iccCopyr = "";
+    }
+
+    cmsFloat64Number iccVersion = cmsGetProfileVersion(cmsprofile);
+    char *iccType = "";
+    if(cmsIsMatrixShaper(cmsprofile)) iccType = "Matrix";
+    else if(cmsIsCLUT(cmsprofile, INTENT_PERCEPTUAL, LCMS_USED_AS_INPUT)) iccType = "LUT";
+
+    char *tooltip = g_strdup_printf(_("embedded ICC profile properties:\n\n"
```

Comment:
You probably want to use `g_markup_printf_escaped` since any of the strings could contain markup characters.

Stupid question: could this be shown in the "image information" module instead?

Reply:
In general you're right, uncontrolled input is better escaped... I just couldn't see how the icc description (actually the profile name) or the copyright text could contain markup.

> Stupid question: could this be shown in the "image information" module instead?

It could be be. It's just a matter of using screen real estate effectively.

Reply:
If it is guaranteed not to contain slashes or < / > (I'm not sure if that's all) then you should be fine. But why take the risk with input that's out of your control? This is what the dedicated function is for; you are saying I'm feeding this string into a markup sensitive environment. Otherwise set non-markup tooltip...

Reply:
> But why take the risk with input that's out of your control?

You are right! I'll make that change.

Reply:
Thanks, done!